packages/logger-pino/src/index.ts

```diff
@@ -0,0 +1,125 @@
+import type {
+  LazyLoggerMessage,
+  Logger as MeshLogger,
+} from '@graphql-mesh/types';
+import { LogLevel } from '@graphql-mesh/utils';
+import type pino from 'pino';
+
+type PinoWithChild = pino.BaseLogger & {
+  child: (meta: any) => PinoWithChild;
+};
+
+function prepareArgs(messageArgs: LazyLoggerMessage[]): Parameters<pino.LogFn> {
+  const flattenedMessageArgs = messageArgs
+    .flat(Infinity)
+    .flatMap((messageArg) => {
+      if (typeof messageArg === 'function') {
+        messageArg = messageArg();
+      }
+      if (messageArg?.toJSON) {
+        messageArg = messageArg.toJSON();
+      }
+      if (messageArg instanceof AggregateError) {
+        return messageArg.errors;
+      }
```

Comment:
Why do we want to flatten the list of errors ? Aren't we loosing the message of the AggregatedError ?

Reply:
We lose it but it is usually a concatenated version of child errors or a message like `Failed with multiple errors` or there might be no message as the message property of AggregateError is optional.
And keeping this kind of messages there is mostly more confusing than being helpful.